// @tanstack/router-generator/dist/cjs/template.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
function fillTemplate(template, values) {
  return template.replace(
    /%%(\w+)%%/g,
    (_, key) => values[key] || ""
  );
}
function getTargetTemplate(target) {
  switch (target) {
    // TODO: Remove this disabled eslint rule when more target types are added.
    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
    case "react":
      return {
        fullPkg: "@tanstack/react-router",
        subPkg: "react-router",
        rootRoute: {
          template: () => [
            'import * as React from "react"\n',
            "%%tsrImports%%",
            "\n\n",
            "%%tsrExportStart%%{\n component: RootComponent\n }%%tsrExportEnd%%\n\n",
            'function RootComponent() { return (<React.Fragment><div>Hello "%%tsrPath%%"!</div><Outlet /></React.Fragment>) };\n'
          ].join(""),
          imports: {
            tsrImports: () => "import { Outlet, createRootRoute } from '@tanstack/react-router';",
            tsrExportStart: () => "export const Route = createRootRoute(",
            tsrExportEnd: () => ");"
          }
        },
        route: {
          template: () => [
            "%%tsrImports%%",
            "\n\n",
            "%%tsrExportStart%%{\n component: RouteComponent\n }%%tsrExportEnd%%\n\n",
            'function RouteComponent() { return <div>Hello "%%tsrPath%%"!</div> };\n'
          ].join(""),
          imports: {
            tsrImports: () => "import { createFileRoute } from '@tanstack/react-router';",
            tsrExportStart: (routePath) => `export const Route = createFileRoute('${routePath}')(`,
            tsrExportEnd: () => ");"
          }
        },
        lazyRoute: {
          template: () => [
            "%%tsrImports%%",
            "\n\n",
            "%%tsrExportStart%%{\n component: RouteComponent\n }%%tsrExportEnd%%\n\n",
            'function RouteComponent() { return <div>Hello "%%tsrPath%%"!</div> };\n'
          ].join(""),
          imports: {
            tsrImports: () => "import { createLazyFileRoute } from '@tanstack/react-router';",
            tsrExportStart: (routePath) => `export const Route = createLazyFileRoute('${routePath}')(`,
            tsrExportEnd: () => ");"
          }
        }
      };
    default:
      throw new Error(`router-generator: Unknown target type: ${target}`);
  }
}
const defaultAPIRouteTemplate = [
  'import { json } from "@tanstack/start";\n',
  "%%tsrImports%%",
  "\n\n",
  `%%tsrExportStart%%{ GET: ({ request, params }) => { return json({ message:'Hello "%%tsrPath%%"!' }) }}%%tsrExportEnd%%
`
].join("");
exports.defaultAPIRouteTemplate = defaultAPIRouteTemplate;
exports.fillTemplate = fillTemplate;
exports.getTargetTemplate = getTargetTemplate;
